### ISSUE TITLE
Removing cloud provider dependencies to kubernetes/kubernetes 

### ISSUE THREAD

**Issue by andrewsykim** [kubernetes/kubernetes#69585]
**Is this a BUG REPORT or FEATURE REQUEST?**:
/kind feature

As part of a long running initiative to remove cloud providers out of kubernetes/kubernetes, it's required to remove internal dependencies to k8s.io/kubernetes so we can migrate them into a staging directory (see [KEP-removing-in-tree-providers](https://github.com/kubernetes/enhancements/blob/master/keps/sig-cloud-provider/20190125-removing-in-tree-providers.md)). The following dependencies need to be removed from `k8s.io/kubernetes/pkg/cloudprovider/providers`:

Dependency checklist:
- [x] `pkg/api/legacyscheme` (@andrewsykim)
- [x] `pkg/api/service` (@andrewsykim)
- [X] `pkg/api/v1/pod` (@andrewsykim)
- [x] `pkg/api/v1/service` (@andrewsykim)
- [X] `pkg/apis/apps` (@andrewsykim)
- [X] `pkg/apis/autoscaling` (@andrewsykim)
- [x] `pkg/apis/core` (@islinwb, @dims)
- [x] `pkg/apis/core/helper` (@islinwb, @dims)
- [X] `pkg/apis/core/install` (@andrewsykim)
- [X] `pkg/apis/core/pods` (@andrewsykim)
- [x] `pkg/apis/core/v1` (@islinwb, @dims)
- [x] `pkg/apis/core/v1/helper` (@islinwb, @andrewsykim)
- [X] `pkg/apis/core/validation` (@andrewsykim)
- [X] `pkg/apis/scheduling` (@andrewsykim)
- [X] `pkg/capabilities` (@andrewsykim)
- [X] `pkg/controller` (@stewart-yu)
- [x] `pkg/credentialprovider` (@tiffanyfay)
- [x] `pkg/credentialprovider/aws` (@tiffanyfay)
- [X] `pkg/cloudprovider` (@cheftako) 
- [x] `pkg/features` (@andrewsykim)
- [X] `pkg/fieldpath` (@andrewsykim)
- [x] `pkg/kubelet/apis` (@islinwb , @dims)
- [x] `pkg/version` (@andrewsykim) 
- [X] `pkg/kubelet/types`(@andrewsykim)
- [X] `pkg/master/ports` (@andrewsykim)
- [X] `pkg/security/apparmor`(@andrewsykim)
- [X] `pkg/serviceaccount` (@andrewsykim)
- [x] `pkg/util/keymutex` (@dims)
- [X] `pkg/util/hash` (@andrewsykim)
- [x] `pkg/util/file` (@mcrute) 
- [x] `pkg/util/net/sets` (@mcrute, @ashishranjan738)
- [x] `pkg/util/strings` (@dims)
- [X] `pkg/util/taints` (@stewart-yu)
- [x] `pkg/volume` (@islinwb, @andrewsykim)
- [x] `pkg/volume/util` (@islinwb, @andrewsykim)
- [x] `pkg/volume/util/fs` (@andrewsykim)
- [x] `pkg/volume/util/recyclerclient` (@andrewsykim)
- [x] `pkg/volume/util/types` (@andrewsykim)
- [x] `pkg/volume/util/volumepathhandler` (@andrewsykim)

/sig cloud-provider

**Comment by andrewsykim** [comment]
@kubernetes/sig-cloud-provider @cheftako @dims @d-nishi @frapposelli @hogepodge @justaugustus @mcrute

**Comment by andrewsykim** [comment]
Started work on removing `pkg/util/net/sets`, `pkg/api/v1/service` and `pkg/api/service`

**Comment by islinwb** [comment]
Hi! @andrewsykim I would like to take work on `pkg/apis/core/v1/helper`, `pkg/version`, `pkg/volume`  and    `pkg/volume/util`.

**Comment by stewart-yu** [comment]
`pkg/controller`  
/assign

**Comment by yastij** [comment]
Taking care of `pkg/credentialprovider/aws`

**Comment by stewart-yu** [comment]
@andrewsykim  If i'm think we should determinate some referred packages where to place, istead of copy one for `cloudprovider`. WDYT?

**Comment by andrewsykim** [comment]
@stewart-yu I'm not sure I understand your question, can you clarify a bit please?

**Comment by andrewsykim** [comment]
Ah do you mean we should figure out where we want these dependencies moved before hand?

**Comment by andrewsykim** [comment]
related: we need to have a discussion around where it's appropriate to duplicate code.

**Comment by stewart-yu** [comment]
> Ah do you mean we should figure out where we want these dependencies moved before hand?

Yes

**Comment by andrewsykim** [comment]
Good thinking @stewart-yu! I'm in the process of trying to formailize a meeting for this effort (https://github.com/kubernetes/community/pull/2740). Maybe that's a good place to start, otherwise I'm happy to take suggestions and chat about it here :)

**Comment by mcrute** [comment]
I'll grab `pkg/util/mount`

**Comment by cheftako** [comment]
Hi,
I'd be happy to pick up 
- pkg/master/ports
- pkg/util/node
-pkg/util/parsers

**Comment by cheftako** [comment]
@d-nishi of course :D

**Comment by msau42** [comment]
@mcrute this pr is related to the mount pkg work: https://github.com/kubernetes/kubernetes/pull/68513

**Comment by andrewsykim** [comment]
FYI @mcrute will be doing vendor updates for `k8s.io/utils`

**Comment by dims** [comment]
@andrewsykim please put me down for `pkg/util/keymutex` && `pkg/util/strings `

**Comment by andrewsykim** [comment]
@mcrute assigned `pkg/api/service`, `pkg/api/v1/service` and `pkg/util/net/sets` to you as per our discussion earlier :)

**Comment by andrewsykim** [comment]
For anyone else who has assigned dependencies and have not opened PRs yet, can you provide an estimate for when you can get around to those? If you have a lot on your plate we can reassign to others who have cycles right now

**Comment by mcrute** [comment]
`pkg/util/mount` is blocked on kubernetes/utils#51 but provided that's shipped early next week ETA for that is also next week (w/e Oct. 22).

The remainder I'm anticipating having PR-ready by next week (w/e Oct 22).

**Comment by mcrute** [comment]
@dims yeah I saw that after I had submitted the PRs to move file, io, and nsenter over. I think in all that PR is too big and includes a bunch of things that are already merged into utils. Also the last comment ~4 months ago is that the mount interface needs restructured; the PR wasn't touched since then. I think our best bet is to finish merging the prerequisites and then take another look at moving just mount. We may as well close kubernetes/utils#38 for now. WDYT?

**Comment by dims** [comment]
sounds good @mcrute

**Comment by mcrute** [comment]
@yastij are you still working on `pkg/crednetialprovider/aws`? Mind if I take that?

**Comment by yastij** [comment]
Didn’t start yet, feel free if you can send a PR before the end of the week

**Comment by mcrute** [comment]
I will post a PR either today or tomorrow that breaks the dependency between the AWS cloud provider and the AWS credential provider. Once that PR is merged there should be no need to migrate `pkg/credentialprovider/aws` (which would in-turn require migrating `pkg/credentialprovider`).

**Comment by yastij** [comment]
SGTM

**Comment by luxas** [comment]
FYI, this KEP that I'm writing is probably relevant to this issue: https://docs.google.com/document/d/1nZnzJD9dC0xrtEla2Xa-J6zobbC9oltdHIJ3KKSSIhk/edit#heading=h.5mlljjl6ndja

**Comment by andrewsykim** [comment]
Update on this effort:

There are two pending PRs at the moment: https://github.com/kubernetes/kubernetes/pull/68513 & https://github.com/kubernetes/kubernetes/pull/70675. It's unlikely those PRs will make it in for v1.14 due to the complex nature of trying to refactor that code out of the cloud providers. Hoping they can make it in early in the v1.15 cycle.

**Comment by tallclair** [comment]
FYI: https://github.com/google/cadvisor/pull/2213

**Comment by andrewsykim** [comment]
This is done now given we're not going forward with staging the open stack provider which contained the last internal dependency.
